src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.cs

```diff
@@ -0,0 +1,2718 @@
+        return /*<bind>*/~i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerBitwiseNegation) (OperationKind.UnaryOperatorExpression, Type: System.Int32)
+  IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Int32)
+    ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Char)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Type_BitwiseNot_System_Decimal()
+        {
+            string source = @"
+class A
+{
+    System.Decimal Method()
+    {
+        System.Decimal i = default(System.Decimal);
+        return /*<bind>*/~i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Decimal)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Type_BitwiseNot_System_Single()
+        {
+            string source = @"
+class A
+{
+    System.Single Method()
+    {
+        System.Single i = default(System.Single);
+        return /*<bind>*/~i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Single)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Type_BitwiseNot_System_Double()
+        {
+            string source = @"
+class A
+{
+    System.Double Method()
+    {
+        System.Double i = default(System.Double);
+        return /*<bind>*/~i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Double)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_Type_BitwiseNot_System_Boolean()
+        {
+            string source = @"
+class A
+{
+    System.Boolean Method()
+    {
+        System.Boolean i = default(System.Boolean);
+        return /*<bind>*/~i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
```

Comment:
not sure in C#, how I can get BooleanBitwiseNegation - http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/IExpression.cs,529

which seems possible according to this
http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/BoundTree/Expression.cs,2452